src/algebra/continued_fractions/definitions/basic.lean

```diff
@@ -0,0 +1,321 @@
+{g : generalized_continued_fraction α // g.is_simple_continued_fraction}
+
+variable {α}
+/- Interlude: define some expected coercions. -/
+namespace simple_continued_fraction
+open generalized_continued_fraction as gcf
+open simple_continued_fraction as scf
+variable [has_one α]
+
+/-- Lift a scf to a gcf using the inclusion map. -/
+instance has_coe_to_generalized_continued_fraction : has_coe (scf α) (gcf α) :=
+by {unfold scf, apply_instance}
+
+@[simp, elim_cast]
+lemma coe_to_generalized_continued_fraction {s : scf α} : (↑s : gcf α) = s.val := rfl
+
+end simple_continued_fraction
+
+/-
+Next, we define (regular) continued fractions. A *(regular) continued fraction* ((r)cf) is a simple
+continued fraction whose partial denominators `bᵢ` are all positive, i.e. `0 < bᵢ`.
+-/
+
+/--
+A simple continued fraction is a (regular) continued fraction ((r)cf) if all partial denominators
+are positive.
+-/
+def simple_continued_fraction.is_regular_continued_fraction [has_one α] [has_zero α] [has_lt α]
+(s : simple_continued_fraction α) :
+  Prop :=
+∀ (n : ℕ) (bₙ : α),
+  (↑s : generalized_continued_fraction α).partial_denominators.nth n = some bₙ → 0 < bₙ
+
+variable (α)
+
+/--
+A (regular) continued fraction ((r)cf) is a simple continued fraction (scf) whose partial
+denominators are all positive. It is the subtype of scfs that satisfy
+`simple_continued_fraction.is_regular_continued_fraction`.
+ -/
+def continued_fraction [has_one α] [has_zero α] [has_lt α] :=
+{s : simple_continued_fraction α // s.is_regular_continued_fraction}
+
+variable {α}
+
+/- Interlude: define some expected coercions. -/
+namespace continued_fraction
+open generalized_continued_fraction as gcf
+open simple_continued_fraction as scf
+open continued_fraction as cf
+variables [has_one α] [has_zero α] [has_lt α]
+
+/-- Lift a cf to a scf using the inclusion map. -/
+instance has_coe_to_simple_continued_fraction : has_coe (cf α) (scf α) :=
+by {unfold cf, apply_instance}
+
+@[simp, elim_cast]
+lemma coe_to_simple_continued_fraction {c : cf α} : (↑c : scf α) = c.val := rfl
+
+/-- Lift a cf to a scf using the inclusion map. -/
+instance has_coe_to_generalized_continued_fraction : has_coe (cf α) (gcf α) := ⟨λ c, ↑(↑c : scf α)⟩
+
+@[simp, elim_cast]
+lemma coe_to_generalized_continued_fraction {c : cf α} : (↑c : gcf α) = c.val := rfl
+
+end continued_fraction
+
+/-
+We now define how to compute the convergents of a gcf. There are two standard ways to do this:
+directly evaluating the (infinite) fraction described by the gcf or using a recurrence relation.
+For (r)cfs, these computations are equivalent as shown in
+`algebra.continued_fractions.theorems.convergents_equiv`.
+-/
+namespace generalized_continued_fraction
+open generalized_continued_fraction as gcf
+
+-- Fix a division ring for the computations.
+variable [division_ring α]
+
+/-
+We start with the definition of the recurrence relation. Given a gcf `g`, for all `n ≥ 1`, we define
```

Comment:
I don't understand how the comment here fits with the notation at the beginning of the file. There, your continued fraction is `h + a_0/(b_0+ a_1/(b_1 + ...)`. So, starting first with `h` and then at index `0`. You should probably start at index `1` in the comments at the beginning of the file, and use `A_0 = h` now.

Reply:
Fixed.